index.js

```diff
@@ -21,60 +21,54 @@ const defaultPageResolver = async (_, pages, emojiList, currentPageIndex, reacti
 	return newPage;
 };
 
-const defaultFooterResolver = (currentPageIndex, pagesLength) =>`Page ${currentPageIndex + 1} / ${pagesLength}`;
+const defaultFooterResolver = (currentPageIndex, pagesLength) => `Page ${currentPageIndex + 1} / ${pagesLength}`;
+
+const defaultSendMessage = async (message, pageEmbed) => await message.channel.send(pageEmbed);
```

Comment:
Due to the way Promises work, you don't even have to mark this as `async` and use an explicit `await`!
```ts
async function defaultSendMessage(message: Message, pageEmbed: MessageEmbed): Promise<Message> {
    return await message.channel.send(pageEmbed);
}
```
And
```ts
function defaultSendMessage(message: Message, pageEmbed: MessageEmbed): Promise<Message> {
    return message.channel.send(pageEmbed);
}
```
End up being pretty much exactly the same, as shown via the return type. The only thing you're doing in the first one is unboxing the `Promise<Message>` into the inner `Message` and then reboxing it into another `Promise` due to the `async` function, and in the second, you're just directly returning that `Promise<Message>`. They should both probably return the same error stack should something reject, but I could double check if you'd like; I know that discord.js made its error messages significantly more readable in v12 when you explicitly awaited.

Reply:
Ah right, duh, it's being awaited already at the top level! I just keep coming back to this after ~9 hours of work.